src/mds/CInode.cc

```diff
@@ -3837,6 +3839,25 @@ void CInode::validate_disk_state(CInode::validated_data *results,
         }
       }
 next:
+      if (1) {
+	      std::stringstream ds;
+	      inode_t& inode = in->inode;
+	      JSONFormatter *f = new JSONFormatter();
+              MDCache *mdcache = in->mdcache;
+
+	      mdcache->mds->inotable->dump(f);
+	      f->flush(ds);
+	      dout(0) << "scrub: inotable: adding ino = " << inode.ino <<dendl;
+	      dout(0) << "scrub: inotable status: before = "<< ds.str() << dendl;
+	      mdcache->mds->inotable->repair_inotable(inode.ino);
```

Comment:
You're returning a bool here but then ignoring it: the following save() should be conditional on the result of this